### PR TITLE
Test Buidler EVM with multiple compilers (0.6.3)

### PR DESCRIPTION
**Pull request by fvictorio**
Continuation of #544.

To test this, use:

```
  {
    solidityVersion: "0.6.3",
    compilerPath: "soljson-v0.6.3+commit.8dda9521.js",
    withOptimizations: false,
    runs: 200,
  },
```

as the `CompilerOption` for 0.6.

Not all tests are working, there are five of them that are being skipped. I will work on them next.

### CODE REVIEW

**Comment by alcuadrado** [comment]
I rebased your branch because it was impossible to review this PR otherwise.

File: packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts [hunk 1]
```diff
@@ -1099,9 +1124,14 @@ export class SolidityTracer {
       }
 
       const inst = trace.bytecode.getInstruction(step.pc);
+
       if (inst.location !== undefined) {
         return i;
       }
+
+      if (stopOnLastJump && inst.opcode === Opcode.JUMPI) {
```
Comment (alcuadrado) [inline packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts]: Why `JUMPI`?

Comment (alcuadrado) [reply packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts]: I think you want `inst.jumpType === JumpType.INTO_FUNCTION`

File: packages/buidler-core/src/internal/buidler-evm/stack-traces/compiler-types.ts [hunk 2]
```diff
@@ -49,4 +49,7 @@ export interface CompilerOutputBytecode {
       [libraryName: string]: Array<{ start: 0; length: 20 }>;
     };
   };
+  immutableReferences?: {
```
Comment (alcuadrado) [inline packages/buidler-core/src/internal/buidler-evm/stack-traces/compiler-types.ts:52]: I didn't know about this! 🥳 This is great news! Will make everything easier.

I compiled a few contracts and couldn't find it 😅

File: packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts (hunk 1 above)
Comment (fvictorio) [reply packages/buidler-core/src/internal/buidler-evm/stack-traces/solidityTracer.ts]: I'm... not really sure, I'm pretty much playing it by ear here. But if I change it to `inst.jumpType === JumpType.INTO_FUNCTION` there are five tests that stop working:

- `0_6/call-message/external-calls/non-contract-account-called`
- `0_6/create-message/external-calls/non-contract-account-called`
- `0_6/create-message/no-calls/nonpayable-with-wrong-args`
- `0_6/create-message/no-calls/payable-with-wrong-args`
- `0_6/create-message/no-calls/payable-with-wrong-args-and-value`

To use the last one as example, the test expects a `INVALID_PARAMS_ERROR` but gets a `REVERT_ERROR` in the constructor instead. The last part of the trace looks like this:

```
    ...
    019   ADD                                      c.sol:C:   -  41:40
    020   PUSH1 0x40                               c.sol:C:   -  41:40
    022   MSTORE                                   c.sol:C:   -  41:40
    023   PUSH1 0x20                               
    025   DUP2                                     
    026   LT                                       
    027   ISZERO                                   
    028   PUSH1 0x23                               
    030   JUMPI (INTERNAL_JUMP)            
    031   PUSH1 0x00                               
    033   DUP1                                     
    034   REVERT                                   
```

**Comment by alcuadrado** [comment]
Fixes #596

File: packages/buidler-core/src/internal/buidler-evm/stack-traces/solidity-errors.ts [hunk 3]
```diff
@@ -208,9 +210,17 @@ function getMessageFromLastStackTraceEntry(
...
+    case StackTraceEntryType.FALLBACK_NOT_PAYABLE_AND_NO_RECEIVE_ERROR:
+      return `Transaction reverted: there's no receive function, fallback function is not payable and was called with value ${stackTraceEntry.value.toString(
+        10
+      )}`;
+
     case StackTraceEntryType.UNRECOGNIZED_FUNCTION_WITHOUT_FALLBACK_ERROR:
       return `Transaction reverted: function selector was not recognized and there's no fallback function`;
 
+    case StackTraceEntryType.MISSING_FALLBACK_OR_RECEIVE_ERROR:
+      return `Transaction reverted: function selector was not recognized and there's no fallback or receive function`;
```
Comment (alcuadrado) [inline packages/buidler-core/src/internal/buidler-evm/stack-traces/solidity-errors.ts:222]: I think it "or" should be "nor"

File: packages/buidler-core/test/internal/buidler-evm/stack-traces/test-files/0_6/create-message/no-calls/immutable-variable/test.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "skip": true,
+  "reason": "This code seems to trigger a bug in the compiler",
```
Comment (alcuadrado) [inline packages/buidler-core/test/internal/buidler-evm/stack-traces/test-files/0_6/create-message/no-calls/immutable-variable/test.json:3]: Can we add a comment or readme with a link to the issue where this is reported?

File: packages/buidler-core/test/internal/buidler-evm/stack-traces/test-files/0_6/call-message/external-calls/call-with-value-without-funds/test.json [hunk 5]
```diff
@@ -1,4 +1,5 @@
 {
+  "solc": "^0.6.2",
```
Comment (alcuadrado) [inline packages/buidler-core/test/internal/buidler-evm/stack-traces/test-files/0_6/call-message/external-calls/call-with-value-without-funds/test.json:2]: I don't understand why this is needed.

Comment (alcuadrado) [reply packages/buidler-core/test/internal/buidler-evm/stack-traces/test-files/0_6/call-message/external-calls/call-with-value-without-funds/test.json:2]: Ohh, thew new syntax!

**Pull request merged**